packages/@aws-cdk/aws-events-targets/test/lambda/lambda.test.ts

```diff
@@ -100,7 +100,27 @@ test('adding same singleton lambda function as target mutiple times creates perm
   }));
 
   // THEN
-  expect(stack).to(countResources('AWS::Lambda::Permission', 1));
+  expect(stack).toCountResources('AWS::Lambda::Permission', 1);
+});
+
+test('lambda handler and cloudwatch event across stacks', () => {
```

Comment:
Should there also be a test for an imported Rule, as that changes the behavior?

Reply:
Why is this any different for imported Rule? An imported rule also follows the same logic.

Reply:
Because of the `if (rule instanceof Construct) {` check?

Reply:
Imported resources are also a `Construct`. They inherit `Resource` which is a subclass of `Construct`.

The `if` statement is needed because `IRule` and `Construct` don't overlap. But so far there is no case when the Rule will not be a `Construct`.